cirq/ops/phased_iswap_gate.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ISWAPPowGate conjugated by tensor product Rz(phi) and Rz(-phi)."""
+
+from typing import List, Union, Sequence, Tuple, Optional
+
+import numpy as np
+import sympy
+
+import cirq
+from cirq import value, protocols
+from cirq._compat import proper_repr
+from cirq.ops import gate_features, raw_types, op_tree, common_gates
+from cirq.value import type_alias
+
+
+@value.value_equality(manual_cls=True)
+class PhasedISwapPowGate(common_gates.ISwapPowGate, gate_features.TwoQubitGate):
+    """Fractional ISWAP conjugated by Z rotations.
+
+    PhasedISwapPowGate with phase_exponent p and exponent t is equivalent to
+    the composition
+
+        (Z^-p ⊗ Z^p) ISWAP^t (Z^p ⊗ Z^-p)
+
+    and is given by the matrix:
+
+        [[1, 0, 0, 0],
+         [0, c, i·s·f, 0],
+         [0, i·s·f*, c, 0],
+         [0, 0, 0, 1]]
+
+    where:
+
+        c = cos(π·t/2)
+        s = sin(π·t/2)
+        f = exp(2πi·p)
+
+    and star indicates complex conjugate.
+    """
+
+    def __init__(self,
+                 *,
+                 phase_exponent: Union[float, sympy.Symbol] = 1 / 4,
```

Comment:
It defaults to the fourth root of the iswap?

Reply:
Quarter of the phase, i.e. the T gate (or its inverse) on each qubit. This is the most interesting case since it gives half a full circuit phase difference between off-diagonal elements of the inner block (i.e. YXXY gate).

Explained in comment.